src/UI/preferences/LinkSettings.qml

```diff
@@ -49,12 +49,70 @@ SettingsPage {
         }
     }
 
+    SettingsGroupLayout {
+        heading: qsTr("NMEA GPS")
+        visible: QGroundControl.settingsManager.autoConnectSettings.autoConnectNmeaPort.visible && QGroundControl.settingsManager.autoConnectSettings.autoConnectNmeaBaud.visible
+
+        LabelledComboBox {
+            id: nmeaPortCombo
+            label: qsTr("Device")
+
+            model: ListModel {}
+
+            onActivated: (index) => {
+                if (index !== -1) {
+                    QGroundControl.settingsManager.autoConnectSettings.autoConnectNmeaPort.value = comboBox.textAt(index);
+                }
+            }
+
+            Component.onCompleted: {
```

Comment:
A ListModel doesn't signal when the mode changes. Which in turns causes the combo box to not recalc sizing. You need to do this:
```
            Component.onCompleted: {
                var model = []

                model.push(qsTr("Disabled"))
                model.push(qsTr("UDP Port"))
                if (QGroundControl.linkManager.serialPorts.length === 0) {
                    model.push(qsTr("Serial <none available>"))
                } else {
                    for (var i in QGroundControl.linkManager.serialPorts) {
                        model.push(QGroundControl.linkManager.serialPorts[i])
                    }
                }
                nmeaPortCombo.model = model

                const index = nmeaPortCombo.comboBox.find(QGroundControl.settingsManager.autoConnectSettings.autoConnectNmeaPort.valueString);
                nmeaPortCombo.currentIndex = index;
            }
```